src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -74,8 +74,16 @@ public SonarWebService(IDownloader downloader, string server, ILogger logger)
 
                 var json = JObject.Parse(contents);
                 var profiles = json["profiles"].Children<JObject>();
-
-                var profile = profiles.SingleOrDefault(p => language.Equals(p["language"].ToString()));
+                JObject profile = null;
+                try
+                {
+                    profile = profiles.SingleOrDefault(p => language.Equals(p["language"].ToString()));
+                }
+                catch (InvalidOperationException) //As we don't have fail-fast policy for unsupported version for now, we should handle gracefully multi-QPs set for a project, here for SQ < 6.7
+                {
+                    logger.LogWarning($"Your project has multiple Quality Profiles defined for language `{language}`, taking the default one.");
+                    profile = profiles.SingleOrDefault(p => language.Equals(p["language"].ToString()) && p["isDefault"].Value<bool>().Equals(true));
+                }
```

Comment:
Separately, I think the analysis should fail at this point. If it doesn't, the project will be analysed with the wrong QP, so even if the user sees the build warning (which is easy to miss) the analysis results will be wrong.

I guess it comes down to whether it is better to stop the analysis, or continue using the wrong configuration, which sounds like a user experience question for the PMs. Is there someone from the PM team involved in the scanner guild? It would make sense if there was since the behaviour of the scanners is fundamental to all of the ALM integration features.